### DISCUSSION TITLE
Discussion: Measures to avoid re-centralization?

### DISCUSSION THREAD

**Discussion by k5123** [mastodon/mastodon#22746] [General]
A thought I came across while watching the influx of posts in federated... 

What measures does Mastodon or any other federated service take to avoid re-centralization due to extremely huge instances or clusters thereof? 

Say Amazon uses their infra to set up the largest Mastodon instance in the world. Potentially, I believe this would go right back to centralized administration. 
Note how I'm saying "administration" because that's where the real crux is with services like this. If a site is large enough, people flock there because they feel it's the "standard", and "everyone is there anyways". And again it's owned by one administration, that can then actively work to diminish outside instances, or outright block them. And you have your walled garden again.

So I'm wondering how could this single-owner-problem be countered? Or maybe I'm seeing a problem where there is none...

**Comment by rbairwell** [comment]
> What measures does Mastodon or any other federated service take to avoid re-centralization due to extremely huge instances or clusters thereof?

Absolutely none at all (especially since Mastodon itself is just a piece of software anyone could run.) It's a bit like the Apache webserver software - anybody has the ability to download and run their own webserver independently of anyone else, but there's also nothing stopping say Amazon convincing everybody to move to "their" Apache-powered web host.

> So I'm wondering how could this single-owner-problem be countered? Or maybe I'm seeing a problem where there is none...

There are already a number of "net-splits" (to borrow terminology from IRC) on the "Mastodon ActivityPub Fediverse Network" which will actually reduce (if not stop centralisation). For example, there is a "well-known" Japanese oriented Mastodon server which "Western" servers won't federate with as the content is "potentially illegal" in those countries, there's "right-wing" site such as Gab and TruthSocial which run the Mastodon software but most of the "main Mastodon network" doesn't communicate/federate with them.

However, the likelihood of every single Mastodon instance merging into one is extremely low - the thing that most server administrators are worried about is the "Embrace, Extend And Extinguish" model. Many many moons ago, "the" instant messaging protocol everybody used was called XMPP (also known as Jabber) - and Google thought it was a very good protocol indeed and so they based Google Talk on it ("Embrace" in 2005). However, they then found it didn't quite do everything they wanted and so they started adding extra functionality to it ("Extend" in 2006) - and since Google was one of the biggest "players" quite a few people needed to use Google's system if they wanted to communicate with other Google users.... And then in 2013 they then cut off XMPP protocol to other servers ("Extinguish") - but nobody really cared as they were all on Google using it. And then they shut down Google Talk.... (Paraphrased from the [Wikipedia](https://en.wikipedia.org/wiki/XMPP#Non-native_deployments) article).

As long as people want to run their own servers and other people want to run them, the Mastodon software and the ActivityPub network will keep running (the same way that IRC - introduced in 1988 - is still available today even though it's been in decline since 2003).
